### DISCUSSION TITLE
Portal Vs fullscreen for the dialog

### DISCUSSION THREAD

**Discussion by Irev-Dev** [tailwindlabs/headlessui#1542] [Help]
Hello,

I have a react component which has two buttons, one opens a headless-ui dialog component to reveal some settings, and the other makes the whole component fullscreen (using the fullscreen api but with [this abstraction](https://www.npmjs.com/package/react-full-screen)) 

![image](https://user-images.githubusercontent.com/29681384/171498269-d9c7e457-1251-497b-bc43-b744c5f048e5.png)

However the problem I'm facing is the dialog stops working when it's in fullscreen, I'm guessing this is because of the use of react portal, and the fact that the dialog is actually located elsewhere in the dom and so it stays underneath the full screen element. Is there a work around for this? a way of dictating where the portal is in the dom or disabling it?

**Comment by HolzmanoLagrene** [comment]
I would be very interested in finding a solution for this!

**Comment by ZahidChandio** [comment]
Can anyone please answer this?

**Comment by jvmatias** [comment]
I fixed this by adding id="headlessui-portal-root" directly to the fullscreen div. Now, when I'm in fullscreen and open the modal, it renders inside the fullscreen element instead of the body.
